### PR TITLE
Add 7.3 release highlights.

### PR DESCRIPTION
**Pull request by jpountz**
Currently:
 - voting-only nodes
 - replication of aliases by CCR
 - fixed memory leak with DLS
 - sql supports frozen indices
 - rare terms agg
 - analyzers reload api
 - memory usage of global ords
 - flattened fields
 - cosine similarity and dot product on vectors
 - prefix and wildcards with intervals
 - data frame pivot transforms
 - outlier detection

### CODE REVIEW

File: docs/reference/release-notes/highlights-7.3.0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Read aliases are now replicated via <<ccr-put-follow,{ccr}>>. Note that write
+aliases are still not replicated since they only make since for indices that
+are being written to while follower indices do not receive direct writes.
+
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+[float]
+==== Memory leak when using templates in document-level security
```
Comment (danielmitterdorfer) [inline docs/reference/release-notes/highlights-7.3.0.asciidoc]: Maybe reword to make clear that the highlight is actually a fix for the potential memory leak? E.g.: "Limit memory usage when ..."?

File: docs/reference/release-notes/highlights-7.3.0.asciidoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+{xpack-ref}/document-level-security.html[Document-level security] was using an
+unbounded cache for the set of visible documents. This could lead to a memory
+leaks when using a templated query as a role query. The cache has been fixed to
+evict based on memorory usage and has a limit of 50MB.
```
Comment (danielmitterdorfer) [inline docs/reference/release-notes/highlights-7.3.0.asciidoc]: Typo: `memorory` -> `memory`

File: docs/reference/release-notes/highlights-7.3.0.asciidoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+distribution of Elasticsearch.
+
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+[float]
+==== Aliases are replicated via {ccr}
+
+Read aliases are now replicated via <<ccr-put-follow,{ccr}>>. Note that write
+aliases are still not replicated since they only make since for indices that
```
Comment (danielmitterdorfer) [inline docs/reference/release-notes/highlights-7.3.0.asciidoc]: "since they only make since" -> "since they only make *sense*"?

File: docs/reference/release-notes/highlights-7.3.0.asciidoc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+are being written to while follower indices do not receive direct writes.
+
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+[float]
+==== Memory leak when using templates in document-level security
+
+{xpack-ref}/document-level-security.html[Document-level security] was using an
+unbounded cache for the set of visible documents. This could lead to a memory
```
Comment (danielmitterdorfer) [inline docs/reference/release-notes/highlights-7.3.0.asciidoc]: "This could lead to a memory leak" (singular instead of plural)

**Review by danielmitterdorfer — COMMENTED** [review]
I left some comments

**Comment by yaronp68** [comment]
@jpountz suggest to add the following items:

- [Support for flattened object fields](https://github.com/elastic/elasticsearch/issues/33003
)
- [Support for intervals query with prefix and wildcard] (https://github.com/elastic/elasticsearch/pull/43635)
- [Predefined vector proximity functions](https://github.com/elastic/elasticsearch/pull/43280)

**Comment by jpountz** [comment]
Thanks @yaronp68!

File: docs/reference/release-notes/highlights-7.3.0.asciidoc [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+//NOTE: The notable-highlights tagged regions are re-used in the
+//Installation and Upgrade Guide
+
+// tag::notable-highlights[]
+[float]
+==== Voting-only master nodes
+
+A new <<voting-only-node,`node.voting-only`>> role has been introduced that
+allows nodes to participate in elections even though they are not eligible.
```
Comment (tvernum) [inline docs/reference/release-notes/highlights-7.3.0.asciidoc]:
```suggestion
allows nodes to participate in elections even though they are not eligible to become the master.
```

**Comment by yaronp68** [comment]
@jpountz forgot to mention a very big one - [support for data frame](https://github.com/elastic/elasticsearch/pull/43544)

**Comment by jpountz** [comment]
@yaronp68 Added.

**Review by yaronp68 — APPROVED** [review]

**Comment by jpountz** [comment]
Thanks @yaronp68 !
